### PR TITLE
Add Tcp IPv6 Metric for Windows container

### PR DESCRIPTION
**Pull request by tekian**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4425)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Internal/Windows/Network/TcpTableInfo.cs [hunk 1]
```diff
@@ -86,9 +136,22 @@ private static IntPtr RetryCalling()
                 $"For more information see https://learn.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-gettcptable");
     }
 
-    private unsafe TcpStateInfo GetSnapshot()
+    private void RefreshSnapshotIfNeeded()
     {
-        var tcpTable = RetryCalling();
+        lock (_lock)
+        {
+            if (_refreshAfter < TimeProvider.GetUtcNow())
```
Comment (tekian) [inline src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Internal/Windows/Network/TcpTableInfo.cs:143]: From `IClock` to `TimeProvider`. @geeknoid

File: src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Internal/Windows/Network/TcpTableInfo.cs [hunk 2]
```diff
@@ -86,9 +136,22 @@ private static IntPtr RetryCalling()
...
+            {
+                _iPv4Snapshot = GetSnapshot();
+                _iPv6Snapshot = GetIPv6Snapshot();
+                _refreshAfter = TimeProvider.GetUtcNow().Add(_samplingInterval);
```
Comment (tekian) [inline src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Internal/Windows/Network/TcpTableInfo.cs:147]: From `IClock` to `TimeProvider`. @geeknoid

**Review by geeknoid — APPROVED** [review]

**Comment by geeknoid** [comment]
@tekian Important: please merge all PRs into the release/8.0 branch instead of main. This is what will be shipped in November. Automation will copy the changes to the main branch automatically, but it will not copy from main into release/8.0 automatically.

**Comment by tekian** [comment]
Merging to `release/8.0` instead.

**Pull request closed without merging**